sphinx/util/cfamily.py

```diff
@@ -134,8 +135,9 @@ def _stringify(self, transform: StringifyTransform) -> str:
         return "[[" + self.arg + "]]"
 
     def describe_signature(self, signode: TextElement) -> None:
-        txt = str(self)
-        signode.append(nodes.Text(txt, txt))
+        signode.append(sphinx.addnodes.desc_sig_punctuation('[[', '[['))
+        signode.append(sphinx.addnodes.desc_sig_keyword(self.arg, self.arg))
+        signode.append(sphinx.addnodes.desc_sig_punctuation(']]', ']]'))
```

Comment:
```suggestion
        signode.append(addnodes.desc_sig_punctuation('[[', '[['))
        signode.append(nodes.Text(self.arg, self.arg))
        signode.append(addnodes.desc_sig_punctuation(']]', ']]'))
```

Reply:
In the near'ish future the attribute handling will be more elaborate (https://github.com/sphinx-doc/sphinx/issues/9904), so  for now let's skip the semantic markup so users don't rely on it.